Common/HipPlatform.cmake

```diff
@@ -0,0 +1,112 @@
+# MIT License
+#
+# Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+cmake_minimum_required(VERSION 3.21...3.27)
+
+include_guard(GLOBAL)
+
+set(ROCM_EXAMPLES_WERROR_DEV OFF CACHE BOOL "Treat CMake developer warnings as errors.")
+
+function(select_gpu_language)
+    if (DEFINED CACHE{GPU_RUNTIME} AND NOT "${ROCM_EXAMPLES_GPU_RUNTIME_DISABLE_WARN}")
+        message(DEPRECATION "GPU_RUNTIME is deprecated, please use ROCM_EXAMPLES_GPU_LANGUAGE to set"
```

Comment:
After this change, we see ~300 cmake warnings. I think we should clean up the remaining examples before finally merging this PR.

Reply:
This is because there are still two phases yet to be implemented until the feature is complete (that is, until the examples can be used with the NVIDIA platform of CMake's hip language support). We can drop this commit for now and push the feature when it is fully implemented if that's better.

Reply:
We can keep the PR open until subsequent parts are done. I just think it would be a bad user experience to anyone trying to build the examples and then seeing all these warnings.

Reply:
Well, the second and third phases of this feature will take quite a long time. I just dropped this commit to not drag this PR further, and we'll upstream it when finished.